### ISSUE TITLE
fidel plugin system

### ISSUE THREAD

**Issue by jgallen23** [jgallen23/fidel#9]
Right now we have events for FidelPostInit, but I think there's probably a better way to extend fidel.  If we build correctly, we can split the features up into their own plugins

**Comment by jgallen23** [comment]
Idea 1 (data-elements example)

``` javascript
fidel.plugin({
  defaults: {
  },
  init: function() {
    this.getDataElements();
  },
  getDataElements: function() {
     this.find('[data-element]').forEach(function(el) {
         self.els[name] = el;
     });
  }
});
```

So fidel would just expose a bare bones amount of functionality, but it could be easily extended.  We would have built in plugins for data elements, events, delegate actions, etc.  And then we could build plugins for integration with bind-to, etc.

**Comment by jgallen23** [comment]
@Belelros @dawnerd let me know what you think

**Comment by Antonio-Laguna** [comment]
I think this is great but, on the other hand I think we would end up bloating the library a little bit just to achieve this and most of the stuff will always be included.

**Comment by jgallen23** [comment]
I'm hoping it wouldn't add too much bloat. Maybe we don't allow options and defaults for plugins, they are just an init, destroy and custom methods. 

I don't think everything will be built in. Here are some ideas for plugins that won't be built in bind-to, template rendering, form validation, hash routing.  

More than anything I think it cleans up the code and gives us the flexibility to build on Fidel without making the core lib too big.

**Comment by Antonio-Laguna** [comment]
Oh, I see. Now it has more sense and I think it would be really great having the ability to hand-pick some pieces of functionality here and there.

**Comment by Antonio-Laguna** [comment]
How this should work?

I imagine plugins being stored as they're declared internally, then on the module definition, we could have an array of plugins to use and we would extend the module with the given plugins.

Do you have anything else in mind?

**Comment by jgallen23** [comment]
Yep. Sounds right. Take a stab at what you are thinking in a branch
